connect/api/src/main/java/org/apache/kafka/connect/sink/SinkConnector.java

```diff
@@ -34,4 +34,12 @@
      */
     public static final String TOPICS_CONFIG = "topics";
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected SinkConnectorContext context() {
```

Comment:
Right now, this method is unused, which means we have no tests for this. The existing `MockConnector` class uses the `context` field from `Connector`, and at a minimum I think it be worthwhile to consider changing that to call the `context()` method. This class is wrapped by `MockSinkConnector` and `MockSourceConnector`, and used in unit tests.

However, it's probably worth adding a few new unit test classes for `SinkConnector` and `SourceConnector` so that we can verify things like `context()` returns the same instance as supplied to `initialize(...)`, that the `context` field also has the same instance as supplied to `initialize(...)`, and that the casting in `SinkConnector.context()` and `SourceConnector.context()` is actually safe.